src/notebooks/controllers/controllerPreferredService.ts

```diff
@@ -280,6 +280,10 @@ export class ControllerPreferredService implements IControllerPreferredService,
                         targetController.id
                     } for document ${getDisplayPath(document.uri)}`
                 );
+                await this.preferredControllers
+                    .get(document)
+                    ?.controller.updateNotebookAffinity(document, NotebookControllerAffinity.Default);
```

Comment:
Oh didn't know that this was necessary, I thought each extension could only contribute one `suggested` controller, i.e. when you suggest one then that would overrride the previously suggested controller.